src/Message/MessageFactoryInterface.php

```diff
@@ -0,0 +1,108 @@
+<?php
+
+/*
+ * This file is part of the Http Adapter package.
+ *
+ * (c) Eric GELOEN <geloen.eric@gmail.com>
+ *
+ * For the full copyright and license information, please read the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Http\Adapter\Message;
+
+use Http\Adapter\HttpAdapterException;
+use Psr\Http\Message\UriInterface;
+use Psr\Http\Message\StreamInterface;
+
+/**
+ * @author GeLo <geloen.eric@gmail.com>
+ */
+interface MessageFactoryInterface
+{
+    /**
+     * Returns the base URI
+     *
+     * @return null|UriInterface
+     */
+    public function getBaseUri();
+
+    /**
+     * Checks if there is a base URI
+     *
+     * @return boolean
+     */
+    public function hasBaseUri();
+
+    /**
+     * Sets the base URI
+     *
+     * @param null|string|UriInterface $baseUri
+     *
+     * @throws HttpAdapterException If the base uri is invalid.
```

Comment:
If it comes from the old package, I'm wrong... By reading the current implementation, this is an exception which comes from `phly/http` so it can't be an `HttpAdapterException`.